### PR TITLE
Add registration endpoints

### LINKED ISSUE #4: Add POST registration endpoints

### PR DESCRIPTION
**Pull request by chrisrohr**
Closes #4

### CODE REVIEW

File: src/main/java/org/kiwiproject/eureka/EurekaServletHandler.java [hunk 1]
```diff
@@ -164,7 +173,6 @@ private int getResponseCodeWithPossibleRetryFailure(ConcurrentHashMap<String, Pa
                 } else {
                     LOG.debug("Reached num times to fail {} for retry key {}, returning success code {}",
                             numTimesToFail, retryKey, successResponseCode);
-                    responseStatusCode = HttpServletResponse.SC_OK;
```
Comment (sleberknight) [inline src/main/java/org/kiwiproject/eureka/EurekaServletHandler.java:167]: Why did you remove this? Especially in light of the log message immediately above it. But it seems that removing it would cause problems, or am I missing something?

Comment (chrisrohr) [reply src/main/java/org/kiwiproject/eureka/EurekaServletHandler.java:167]: The default is setting it to the success code.... this was ignoring the success code being passed in

File: src/test/java/org/kiwiproject/eureka/EurekaServletHandlerTest.java [hunk 2]
```diff
@@ -412,4 +415,93 @@ void shouldReturn500ForStatusOnTriggeredFailure() {
...
+                    .resolveTemplate("appId", APP_NAME)
+                    .request(MediaType.APPLICATION_JSON_TYPE)
+                    .post(json(postBody));
+
+            assertNoContentResponse(response);
+            verify(eurekaServer).registerApplication(isA(InstanceInfo.class));
+        }
+
+        @Test
+        void shouldAttemptToRegisterAndReturn500OnTriggeredFailure() {
```
Comment (sleberknight) [inline src/test/java/org/kiwiproject/eureka/EurekaServletHandlerTest.java:448]: Might want to include an `_` between the `500` and `OnTriggeredFailure` -- I did a double take since I thought it said 5000 🤔

File: src/test/java/org/kiwiproject/eureka/EurekaServletHandlerTest.java [hunk 3]
```diff
@@ -412,4 +415,93 @@ void shouldReturn500ForStatusOnTriggeredFailure() {
...
+                    .resolveTemplate("appId", APP_NAME)
+                    .request(MediaType.APPLICATION_JSON_TYPE)
+                    .post(json(postBody));
+
+            assertInternalServerErrorResponse(response);
+            verifyNoInteractions(eurekaServer);
+        }
+
+        @Test
+        void shouldAttemptToRegisterAndReturn500OnTriggeredFailure_ThenSuccessAfterTriggeredNumber() {
```
Comment (sleberknight) [inline src/test/java/org/kiwiproject/eureka/EurekaServletHandlerTest.java:473]: Ditto

File: src/test/java/org/kiwiproject/eureka/EurekaServletHandlerTest.java [hunk 4]
```diff
@@ -412,4 +415,93 @@ void shouldReturn500ForStatusOnTriggeredFailure() {
...
+                assertInternalServerErrorResponse(response);
+            }
+
+            var response = client.target(server.getURI())
+                    .path("/eureka/v2/apps/{appId}")
+                    .resolveTemplate("appId", APP_NAME)
+                    .request(MediaType.APPLICATION_JSON_TYPE)
+                    .post(json(postBody));
+
+            assertNoContentResponse(response);
```
Comment (sleberknight) [inline src/test/java/org/kiwiproject/eureka/EurekaServletHandlerTest.java:503]: OK so I assume this is testing what I asked about earlier? i.e. that it fails up to a specific number of retries, and then finally returns success???

**Review by sleberknight — CHANGES_REQUESTED** [review]
See comments

File: src/test/java/org/kiwiproject/eureka/EurekaServletHandlerTest.java (hunk 2 above)
Comment (chrisrohr) [reply src/test/java/org/kiwiproject/eureka/EurekaServletHandlerTest.java]: Done

File: src/test/java/org/kiwiproject/eureka/EurekaServletHandlerTest.java (hunk 3 above)
Comment (chrisrohr) [reply src/test/java/org/kiwiproject/eureka/EurekaServletHandlerTest.java]: Done

File: src/test/java/org/kiwiproject/eureka/EurekaServletHandlerTest.java (hunk 4 above)
Comment (chrisrohr) [reply src/test/java/org/kiwiproject/eureka/EurekaServletHandlerTest.java:503]: Correct

File: src/main/java/org/kiwiproject/eureka/EurekaServletHandler.java (hunk 1 above)
Comment (sleberknight) [reply src/main/java/org/kiwiproject/eureka/EurekaServletHandler.java:167]: OK

**Review by sleberknight — APPROVED** [review]

**Review by sleberknight — APPROVED** [review]

**Review by sleberknight — APPROVED** [review]
